src/hero.tsx

```diff
@@ -0,0 +1,213 @@
+    buttonLink?: RendersNullableElement<typeof ButtonLink>;
+};
+
+const HeroContent = ({
+    headline,
+    title,
+    pretitle,
+    description,
+    descriptionLinesMax,
+    extra,
+    button,
+    buttonLink,
+}: HeroContentProps) => {
+    return (
+        <section
+            className={sprinkles({
+                height: '100%',
+                display: 'flex',
+                justifyContent: 'space-between',
+                flexDirection: 'column',
+            })}
+        >
+            <div>
+                <Stack space={16}>
+                    {headline && headline}
+                    <Stack space={8}>
+                        {pretitle && (
+                            <Text3 as="p" regular>
+                                {pretitle}
+                            </Text3>
+                        )}
+                        {title && <Text8 as="h1">{title}</Text8>}
+                    </Stack>
+                    {description && (
+                        <Text3
+                            as="p"
+                            regular
+                            color={vars.colors.textSecondary}
+                            truncate={descriptionLinesMax}
+                        >
+                            {description}
+                        </Text3>
+                    )}
+                </Stack>
+                {extra && <div>{extra}</div>}
+            </div>
+            {(button || buttonLink) && (
+                <div className={styles.actions}>
+                    <ButtonGroup primaryButton={button} link={buttonLink} />
+                </div>
+            )}
+        </section>
+    );
+};
+
+type HeroProps = {
+    height?: string;
+    background?: 'default' | 'alternative' | 'brand' | 'brand-secondary';
+    media: RendersElement<typeof Image> | RendersElement<typeof Video>;
+    headline?: RendersNullableElement<typeof Tag>;
+    pretitle?: string;
+    title?: string;
+    description?: string;
+    descriptionLinesMax?: number;
+    extra?: React.ReactNode;
+    button?: RendersNullableElement<typeof ButtonPrimary>;
+    buttonLink?: RendersNullableElement<typeof ButtonLink>;
+    dataAttributes?: DataAttributes;
+    desktopMediaPosition?: 'left' | 'right';
+};
+
+const BACKGROUND_COLOR = {
+    default: vars.colors.background,
+    alternative: vars.colors.backgroundAlternative,
+    brand: vars.colors.backgroundBrand,
+    'brand-secondary': vars.colors.backgroundBrand, // TODO backgroundBrandAlternative does not exists yet
+};
+
+const Hero = React.forwardRef<HTMLDivElement, HeroProps>(
+    (
+        {height, background = 'default', media, desktopMediaPosition = 'left', dataAttributes, ...rest},
+        ref
+    ) => {
+        const {isTabletOrSmaller} = useScreenSize();
+        const isInsideSlideShow = useIsInsideSlideshowContext();
+        const isInverse = background === 'brand' || background === 'brand-secondary';
+
+        if (isTabletOrSmaller) {
```

Comment:
have you tried to avoid using this hook and use mediaqueries instead?

Reply:
we have two things that make this not possible, first the value of `MediaBorderRadiusProvider` that depends on window size.
Also GridLayout needed in desktop will suppose more code duplication to implement it in Hero
do you have any other idea?

Reply:
ok, let it this way for the moment

Reply:
if we want to improve this in the future, I think we could:

- GridLayout should fallback to 1 col in mobile, so not a big deal, I think
- For BorderRadius we could use a css class + mediaquery to disable/enable border-radius in images depending on this ancestor class. Maybe too convoluted, not sure if worth it